Source/WebCore/rendering/PositionedLayoutConstraints.cpp

```diff
@@ -321,7 +321,13 @@ void PositionedLayoutConstraints::resolvePosition(RenderBox::LogicalExtentComput
     };
 
     // See CSS2 § 10.3.7-8 and 10.6.4-5.
-    auto position = m_insetModifiedContainingRange.min() + usedMarginBefore + alignmentShift();
+    ItemPosition resolvedAlignment = resolveAlignmentValue();
+    auto position = LayoutUnit();
+    if (ItemPosition::AnchorCenter == resolvedAlignment && !m_defaultAnchorBox.get()) {
+        auto itemSize = computedValues.m_extent + usedMarginBefore + usedMarginAfter;
+        position = m_insetModifiedContainingRange.min() + (m_insetModifiedContainingRange.size() - itemSize) / 2;
+    } else
+        position = m_insetModifiedContainingRange.min() + usedMarginBefore + alignmentShift();
```

Comment:
Is there a reason why we're not just making `resolveAlignmentValue()` return `ItemPosition::Center` when there's no `m_defaultAnchorBox`?

Reply:
That seems to be exactly what the spec is calling for, so what are the cases where this approach causes incorrect rendering?